packages/libs/user-datasets/src/lib/Components/UploadForm.tsx

```diff
@@ -651,124 +654,121 @@ function UploadForm({
+                nestedInputObject: contacts,
+                index,
+              });
+              return (
+                <ContactInput
+                  n={index}
+                  name={contact.name}
+                  onAddName={(value: string) => {
+                    const updatedContacts = updateContactsObject(value, 'name');
+                    setContacts(updatedContacts);
+                  }}
+                  email={contact.email}
+                  onAddEmail={(value: string) => {
+                    const updatedContacts = updateContactsObject(
+                      value,
+                      'email'
+                    );
+                    setContacts(updatedContacts);
+                  }}
+                  affiliation={contact.affiliation}
+                  onAddAffiliation={(value: string) => {
+                    const updatedContacts = updateContactsObject(
+                      value,
+                      'affiliation'
+                    );
+                    setContacts(updatedContacts);
+                  }}
+                  city={contact.city}
+                  onAddCity={(value: string) => {
+                    const updatedContacts = updateContactsObject(value, 'city');
+                    setContacts(updatedContacts);
+                  }}
+                  state={contact.state}
+                  onAddState={(value: string) => {
+                    const updatedContacts = updateContactsObject(
+                      value,
+                      'state'
+                    );
+                    setContacts(updatedContacts);
+                  }}
+                  country={contact.country}
+                  onAddCountry={(value: string) => {
+                    const updatedContacts = updateContactsObject(
+                      value,
+                      'country'
+                    );
+                    setContacts(updatedContacts);
+                  }}
+                  address={contact.address}
+                  onAddAddress={(value: string) => {
+                    const updatedContacts = updateContactsObject(
+                      value,
+                      'address'
+                    );
+                    setContacts(updatedContacts);
+                  }}
+                  isPrimary={contact.isPrimary}
+                  onAddIsPrimary={(value: boolean) => {
+                    const updatedContacts = updateContactsObject(
+                      value,
+                      'isPrimary'
+                    );
+                    setContacts(updatedContacts);
+                    return;
+                  }}
+                  onRemoveContact={(
+                    event: React.MouseEvent<HTMLButtonElement>
+                  ) => {
+                    event.preventDefault();
+                    const updatedContacts = [...contacts];
+                    updatedContacts.splice(index, 1);
+                    setContacts(updatedContacts);
```

Comment:
Since these are repeated a few times, we could consider something a bit more concise, that again makes use of the functional updater.

```suggestion
                    setContacts(prev => prev.filter((_, i) => i !== index));
```